http/tls.go

```diff
@@ -36,6 +37,23 @@ func setTLSHeader(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getTLSConfig(setupRootCA bool) *tls.Config {
+	certPEM := config.GetConfig().GetString("agent.X509_cert")
+	keyPEM := config.GetConfig().GetString("agent.X509_key")
+	var tlsConfig *tls.Config = nil
+	if certPEM != "" && keyPEM != "" {
+		tlsConfig = common.SetupTLSClientConfig(certPEM, keyPEM)
+		if setupRootCA {
+			analyzerCertPEM := config.GetConfig().GetString("analyzer.X509_cert")
+			if analyzerCertPEM != "" {
```

Comment:
You changed the original behavior as previously if this field was not set the client was exiting. Any reason for changing this ?

Reply:
I have removed this condition ans will call setup of CA regardless of the values of "analyzer.X509_cert".